### PR TITLE
Fix slicing

### LINKED ISSUE #359: Non-informative error in simulation mode on partly labeled dataset
In the simulation mode, the error being raised on a partly labeled (as well as unlabelled?) dataset is not informative and of the wrong error type. 

```
jonathans-mbp:simulation_fails_on_partly_labeled jonathan$ asreview simulate ace_10.csv -s bug.h5

            _____ _____            _
     /\    / ____|  __ \          (_)
    /  \  | (___ | |__) |_____   ___  _____      __
   / /\ \  \___ \|  _  // _ \ \ / / |/ _ \ \ /\ / /
  / ____ \ ____) | | \ \  __/\ V /| |  __/\ V  V /
 /_/    \_\_____/|_|  \_\___| \_/ |_|\___| \_/\_/

---------------------------------------------------------------------------------
|                                                                                |
|  Welcome to the ASReview Automated Systematic Review software.                 |
|  In this mode the computer will simulate how well the ASReview software        |
|  could have accelerate the systematic review of your dataset.                  |
|  You can sit back and relax while the computer runs this simulation.           |
|                                                                                |
|  GitHub page:        https://github.com/asreview/asreview                      |
|  Questions/remarks:  asreview@uu.nl                                            |
|                                                                                |
---------------------------------------------------------------------------------

WARNING:root:Simulating partial review, ignoring unlabeled papers (n=3.
Traceback (most recent call last):
  File "/Users/jonathan/.pyenv/versions/anaconda3-2018.12/bin/asreview", line 11, in <module>
    sys.exit(main())
  File "/Users/jonathan/.pyenv/versions/anaconda3-2018.12/lib/python3.7/site-packages/asreview/__main__.py", line 40, in main
    entry.load()().execute(sys.argv[2:])
  File "/Users/jonathan/.pyenv/versions/anaconda3-2018.12/lib/python3.7/site-packages/asreview/entry_points/simulate.py", line 34, in execute
    review_simulate(path, **args_dict)
  File "/Users/jonathan/.pyenv/versions/anaconda3-2018.12/lib/python3.7/site-packages/asreview/review/factory.py", line 239, in review_simulate
    review(dataset, *args, mode='simulate', **kwargs)
  File "/Users/jonathan/.pyenv/versions/anaconda3-2018.12/lib/python3.7/site-packages/asreview/review/factory.py", line 222, in review
    reviewer = get_reviewer(*args, mode=mode, model=model, **kwargs)
  File "/Users/jonathan/.pyenv/versions/anaconda3-2018.12/lib/python3.7/site-packages/asreview/review/factory.py", line 197, in get_reviewer
    **kwargs)
  File "/Users/jonathan/.pyenv/versions/anaconda3-2018.12/lib/python3.7/site-packages/asreview/review/simulate.py", line 47, in __init__
    as_data = as_data.slice(labeled_idx)
  File "/Users/jonathan/.pyenv/versions/anaconda3-2018.12/lib/python3.7/site-packages/asreview/data.py", line 179, in slice
    return ASReviewData(self.df[idx], data_name="sliced")
  File "/Users/jonathan/.pyenv/versions/anaconda3-2018.12/lib/python3.7/site-packages/pandas/core/frame.py", line 2806, in __getitem__
    indexer = self.loc._get_listlike_indexer(key, axis=1, raise_missing=True)[1]
  File "/Users/jonathan/.pyenv/versions/anaconda3-2018.12/lib/python3.7/site-packages/pandas/core/indexing.py", line 1553, in _get_listlike_indexer
    keyarr, indexer, o._get_axis_number(axis), raise_missing=raise_missing
  File "/Users/jonathan/.pyenv/versions/anaconda3-2018.12/lib/python3.7/site-packages/pandas/core/indexing.py", line 1640, in _validate_read_indexer
    raise KeyError(f"None of [{key}] are in the [{axis_name}]")
KeyError: "None of [Int64Index([0, 1, 2, 3, 4, 5, 6], dtype='int64')] are in the [columns]"
```

I think the API should raise an informative `ValueError`.  The CLI should print the error without traceback.

### PR DESCRIPTION
**Pull request by qubixes**
Fixes a bug that prevents the simulation of partially labeled datasets.

Resolves #359

### CODE REVIEW

**Comment by J535D165** [comment]
Thank you so much! Can you add a small unit test? 

What is the expected behaviour? It simulates the labeled subset of the dataset? Should a warning be given?

**Comment by qubixes** [comment]
I can see if I can write a small unit test.

A warning is already given, but yes it will simulate the labeled subset.

**Comment by J535D165** [comment]
Thanks for the heads up. Makes sense.

**Comment by qubixes** [comment]
I have also fixed an issue with prior_idx. I think we're good to go.

File: tests/test_simulate.py [hunk 1]
```diff
@@ -167,12 +180,32 @@ def check_state(state):
     assert len(state.get("labels")) == 6
 
 
+def check_partial_state(state):
```
Comment (J535D165) [inline tests/test_simulate.py:183]: Nice!

**Pull request merged**